.github/workflows/install_deps/action.yml

```diff
@@ -8,22 +8,32 @@ inputs:
     description: The version of Python to use in order to run unit tests
     required: false
     default: '3.9'
+  install_numpy_2:
+    description: Indicate if numpy 2.0+ should be installed or not
+    required: false
+    type: boolean
+    default: false
+  numpy_2_version:
+    description: The version of numpy 2.0 to use in order to run unit tests
+    required: false
+    type: string
+    default: '2.0.0'
```

Comment:
Why do we add this to the workflows as opposed to pinning a `numpy` version in `requirements.txt`?

Reply:
If I understand this correctly, this is needed because we are expecting to be able  to test Pennylane both with Numpy 1.x and 2.x, so this action is in preparation to be able to install one or the other version.

Reply:
Exactly that --- we need to run tests on both the 1.x and 2.x series.

Reply:
I believe `requirements.txt` still pins `numpy` to `1.23`. Should we unpin that?

Reply:
We had a conversation about this in a thread with @timmysilv, quoting:
> requirements.txt can serve as a minimum version for dependencies, whereas requirements-ci.txt allows CI to test with the maximum version

So maybe we can change `numpy~=1.23` to `numpy>=1.23` ?

Reply:
@mlxd  Please let me know if you have issues with keeping `numpy` pinned to 1.23 in `requirements.txt`, otherwise we can mark this conversation as resolved